Gateway/Request/CancelDataBuilder.php

```diff
@@ -67,6 +67,13 @@ public function build(array $buildSubject): array
             $payment->getEntityId()
         );
 
+        //Handle the case when we cancel an open offer
+        if (empty($pspReferences)) {
```

Comment:
This implementation doesn't cover partial payments. Let's discuss the approach again.